pkg/kvstore/etcd.go

```diff
@@ -1055,9 +1094,22 @@ reList:
 	recreateWatcher:
 		scopedLog.WithField(fieldRev, nextRev).Debug("Starting to watch a prefix")
 
-		e.limiter.Wait(ctx)
+		rl, err := e.limiter.Wait(ctx)
+		if err != nil {
+			select {
+			case <-e.client.Ctx().Done():
+				return
+			case <-ctx.Done():
+				return
+			default:
+				goto recreateWatcher
```

Comment:
We are missing `rl.Done()` here, right?

Reply:
We don't need to (and cannot) call `rl.Done()` when `Wait()` returns an error, since `lr` is nil in that case.